examples/pngWithRoundedShapes.php

```diff
@@ -0,0 +1,210 @@
+            imagefilledrectangle(
+                $this->image,
+                $x2 - $rectsize,
+                $y1,
+                $x2,
+                $y1 + $rectsize,
+                $this->moduleValues[$M_TYPE]
+            );
+        }
+
+        if ($neighbours & (1 << 5)) { // neighbour снизу
+            // bottom left
+            imagefilledrectangle(
+                $this->image,
+                $x1,
+                $y2 - $rectsize,
+                $x1 + $rectsize,
+                $y2,
+                $this->moduleValues[$M_TYPE]
+            );
+            // bottom right
+            imagefilledrectangle(
+                $this->image,
+                $x2 - $rectsize,
+                $y2 - $rectsize,
+                $x2,
+                $y2,
+                $this->moduleValues[$M_TYPE]
+            );
+        }
+
+        // ---------------------
+        // inner rounding
+        // ---------------------
+
+        if (!$this->matrix->check($x, $y)) {
+
+            if ($neighbours & 1 && $neighbours & (1 << 7) && $neighbours & (1 << 1)) {
+                // top left
+                imagefilledrectangle(
+                    $this->image,
+                    $x1,
+                    $y1,
+                    $x1 + $rectsize,
+                    $y1 + $rectsize,
+                    $this->moduleValues[$M_TYPE | QRMatrix::IS_DARK]
+                );
+            }
+
+            if ($neighbours & (1 << 1) && $neighbours & (1 << 2) && $neighbours & (1 << 3)) {
+                // top right
+                imagefilledrectangle(
+                    $this->image,
+                    $x2 - $rectsize,
+                    $y1,
+                    $x2,
+                    $y1 + $rectsize,
+                    $this->moduleValues[$M_TYPE | QRMatrix::IS_DARK]
+                );
+            }
+
+            if ($neighbours & (1 << 7) && $neighbours & (1 << 6) && $neighbours & (1 << 5)) {
+                // bottom left
+                imagefilledrectangle(
+                    $this->image,
+                    $x1,
+                    $y2 - $rectsize,
+                    $x1 + $rectsize,
+                    $y2,
+                    $this->moduleValues[$M_TYPE | QRMatrix::IS_DARK]
+                );
+            }
+
+            if ($neighbours & (1 << 3) && $neighbours & (1 << 4) && $neighbours & (1 << 5)) {
+                // bottom right
+                imagefilledrectangle(
+                    $this->image,
+                    $x2 - $rectsize,
+                    $y2 - $rectsize,
+                    $x2,
+                    $y2,
+                    $this->moduleValues[$M_TYPE | QRMatrix::IS_DARK]
+                );
+            }
+        }
+
+        imagefilledellipse(
+            $this->image,
+            (int)(($x * $this->scale) + ($this->scale / 2)),
+            (int)(($y * $this->scale) + ($this->scale / 2)),
+            (int)(2 * $this->options->circleRadius * $this->scale),
```

Comment:
`$this->options->circleRadius` creates some funny (probably unintended) effects when it's set to a value other than 0.5 - I think hardcoding 0.5 would be the better option here.